code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,693 @@
+//		EXAMINE PROC		//			This proc is massive, messy, and probably could be handled better.
+//////////////////////////////			Feel free to try cleaning it up if you think of a better way to do it.
+
+/obj/machinery/fishtank/examine(mob/user)
+	..(user)
+	var/examine_message = ""
+	//Approximate water level
+
+	examine_message += "Water level: "
+
+	if(water_level == 0)
+		examine_message += "\The [src] is empty! "
+	else if(water_level < water_capacity * 0.1)
+		examine_message += "\The [src] is nearly empty! "
+	else if(water_level <= water_capacity * 0.25)
+		examine_message += "\The [src] is about one-quarter filled. "
+	else if(water_level <= water_capacity * 0.5)
+		examine_message += "\The [src] is about half filled. "
+	else if(water_level <= water_capacity * 0.75)
+		examine_message += "\The [src] is about three-quarters filled. "
+	else if(water_level < water_capacity)
+		examine_message += "\The [src] is nearly full! "
+	else if(water_level == water_capacity)
+		examine_message += "\The [src] is full! "
+
+	examine_message += "<br>Cleanliness level: "
+
+	//Approximate filth level
+	if(filth_level == 0)
+		examine_message += "[src] is spotless! "
+	else if(filth_level <= 2.5)
+		examine_message += "[src] looks like the glass has been smudged. "
+	else if(filth_level <= 5)					//This is the breeding threshold
+		examine_message += "[src] has some algae growth in it. "
+	else if(filth_level <= 7.5)
+		examine_message += "[src] has a lot of algae growth in it. "
+	else if(filth_level < 10)
+		examine_message += "[src] is getting hard to see into! Someone should clean it soon! "
+	else if(filth_level == 10)
+		examine_message += "[src] is absolutely disgusting! Someone should clean it NOW! "
+
+	examine_message += "<br>Food level: "
+
+	//Approximate food level
+	if(!fish_count)								//Check if there are fish in the tank
+		if(food_level > 0)						//Don't report a tank that has neither fish nor food in it
+			examine_message += "There's some food in [src], but no fish! "
+	else										//We've got fish, report the food level
+		if(food_level == 0)
+			examine_message += "The fish look very hungry! "
+		else if(food_level < 2)
+			examine_message += "The fish are nibbling on the last of their food. "
+		else if(food_level < 10)				//Breeding is possible
+			examine_message += "The fish seem happy! "
+		else if(food_level == 10)
+			examine_message += "There is a solid layer of fish food at the top. "
+
+	//Report the number of harvestable eggs
+	if(egg_count)								//Don't bother if there isn't any eggs
+		examine_message += "<br>There are [egg_count] eggs able to be harvested! "
+
+	examine_message += "<br>"
+
+	//Report the number and types of live fish if there is water in the tank
+	if(fish_count == 0)
+		examine_message += "\The [src] doesn't contain any live fish. "
+	else
+		//Build a message reporting the types of fish
+		var/fish_num = fish_count
+		var/message = "You spot "
+		while(fish_num > 0)
```

Comment:
this has to be the worst implementation imaginable

use a for each element loop instead